sklearn/metrics/_scorer.py

```diff
@@ -352,7 +371,7 @@ def get_scorer(scoring):
                              'to get valid options.' % scoring)
     else:
         scorer = scoring
-    return scorer
+    return deepcopy(scorer)
```

Comment:
This is a more impact behavior change when `scoring` is a callable. I am okay with it tho.